jsonparse.js

```diff
@@ -408,6 +415,34 @@ proto.onToken = function (token, value) {
   }
 };
 
+/**
+ * determines if precision is lost when parsing the JSON representation of a number to a javascript number
+ * @param {string} jsonNumber
+ * @param {number} javascriptNumber
+ * @return {boolean}
+ */
+function precisionWasLost (jsonNumber, javascriptNumber) {
+  return getDigitsSpecifyingPrecision(jsonNumber) !== getDigitsSpecifyingPrecision(JSON.stringify(javascriptNumber))
+}
+
+/**
+ * Takes in a string represention of a number and returns a simplified version of the significand
+ * as a string.
+ *
+ * The simplified significand only contains the digits and we remove any trailing zeros or decimal
+ * information.
+ * @param {string} numberString
+ * @return {string}
+ */
+function getDigitsSpecifyingPrecision(numberString) {
+  return numberString
+    .replace(/\./, '') // drops the dot if it exists
+    .replace(/^-?0?/, '') // drops leading negative sign and and leading 0's if present
```

Comment:
Does this handle multiple leading zeroes? Example: `'00123'`.

Reply:
I believe it will; however, we decided that we were only going to support what was supported by the JSON specification, so we will not explicitly support that case anyway